Core/DISET/private/Service.py

```diff
@@ -110,7 +121,13 @@ def initialize( self ):
                             }
     #Call static initialization function
     try:
-      self._handler[ 'class' ]._rh__initializeClass( dict( self._serviceInfoDict ),
+      if self.activityMonitoring:
+        self._handler[ 'class' ]._rh__initializeClass( dict( self._serviceInfoDict ),
+                                                     self._lockManager,
+                                                     self._msgBroker,
+                                                     self.log )
```

Comment:
Did you check how this last parameter is used in the `Requesthandler` class ? since your `self.log` and `self._monitor` do not have the same interface, I doubt the `RequestHandler` will be happy

Reply:
Yes I check the method inside `RequestHandler` it was being used here `srv_getMonitor` just to return `cls.__monitor` so didn't understand what to pass there as we are not initializing `self._monitor` when `activityMonitoring` flag is on.

Reply:
Before we were sending the `MonitoringClient` object, so I assume that it would make more sense to send the `MonitoringReporter` object, don't you think ? 
Also, the `RequestHandler` is inherited by all DIRAC services, so it might be used there